api/main.py

```diff
@@ -147,10 +147,22 @@ def get_password_hash(password: Password):
 # -----------------------------------------------------------------------------
 # Nodes
 
-@app.get('/node/{node_id}', response_model=Node)
-async def get_node(node_id: str):
+@app.get('/node/{node_id}')
+async def get_node(node_id: str, kind: str = "node"):
     """Get node information from the provided node id"""
-    return await db.find_by_id(Node, node_id)
+    try:
+        model = db.get_model_from_kind(kind)
```

Comment:
If `kind` was a type with a finite number of possibilities (basically an `Enum`) then I think we could skip this validation as it would be done by Pydantic directly.

Reply:
Yes, having `kind` as `Enum` seems fair.

Reply:
What if a user has stored a Node object with `kind=regression`? Shouldn't we put the validation?

Reply:
Yes I think each model should have only one valid `kind` value. It's a way to "downcast" when going through a series of Node objects. In fact, maybe we shouldn't allow users to submit a value for `kind` and it should always be set with a fixed value directly in each model (like the `created` field I believe).

Reply:
Yes, the fixed value will resolve the issue.